Libraries/MLXLLM/Models/Gemma2.swift

```diff
@@ -288,3 +289,121 @@ extension Gemma2Model: LoRAModel {
         model.layers.map { ($0.attention, ["q_proj", "v_proj"]) }
     }
 }
+
+// TODO - notes
+//
+// - make UnaryLayer extend Module
+// - make a Quantized protocol that provides the groupSize and bits
+// - make the QuantizedLinear shape produce the expanded shape
+// - make `items()` open
+// - make `updateModule(key:_:)` open
+// - make `noGrad` overridable (turn into function?)
+//
+// - evaluation and training should work as expected
+// - this flattens the weights and modules into one layer
+//      - to match the normal lora implementation
+//      - see items() and updateModule()
+
+// TODO: make UnaryLayer extend Module
+public protocol UnaryLayer2: Module {
+    func callAsFunction(_ x: MLXArray) -> MLXArray
+}
+
+/// LoRA layer that can wrap any UnaryLayer
+class LoRA: Module, UnaryLayer2 {
```

Comment:
For reference here is the current implementation of LoRA:

- https://github.com/ml-explore/mlx-swift-examples/blob/main/Libraries/MLXLMCommon/Lora.swift#L71